src/AutoRest.CSharp/Common/Output/Models/MethodSignature.cs

```diff
@@ -32,5 +33,6 @@ public MethodSignature(string name, string? description, string modifiers, CShar
         public FormattableString? ReturnDescription { get; }
         public Parameter[] Parameters { get; }
         public MethodSignature? BaseMethod { get; }
+        public bool UseBaseKeyword { get; }
```

Comment:
It's strange to use BaseMethod for a non-base method.

Reply:
Yeah, maybe `BaseMethod` shouldn't be part of `MethodSignature` since it isn't actually a method signature. Let me check how hard is it to move it out,

Reply:
@AlexanderSher how about introducing a `MethodInvocation` type? The `BaseMethod` here is actually an invocation of a method. I once introduced a type `ParameterInvocation` here: https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Output/Models/ParameterInvocation.cs

Reply:
I've split the `ConstructorSignature` and `MethodSignature` (should've done it in original PR): https://github.com/Azure/autorest.csharp/pull/1903

Reply:
Discussed offline, #1903 will be applied after this PR